docs/runbook.md

```diff
@@ -13,103 +13,317 @@
+
+Recommend to use secret file or env variable. It's not recommend to specify secret value.
+
 #### file
 
-#### name
+type: list of str, optional, default: empty
 
-### value
+Specify path of other yml files which define variables, how file and value work together.
 
-### artifact
+```
+variable:
+  - file: ../variable_database.yml
+    is_secret: true
+  - file: ../../.vscode/secret.yml
+    is_secret: true
+```
 
 #### name
 
-#### locations
+type: str, optional, default is empty.
 
-##### type
+Variable name.
 
-##### path
+#### value
+
+type: str, optional, default is empty
+
+Value of the paired variable.
 
+```
+variable:
+  - name: name
+    value: Azure Test
+  - name: test_project
+    value: Test Project 1
+  - name: test_pass
+    value: Test Pass 1
+  - name: test_tag
+    value: round 1
+```
 
 ### notifier
 
+Receive messages during the test run and output them somewhere. For more info, please refer [Notifier](https://github.com/microsoft/lisa/blob/main/docs/extension.md#notifier).
+
 #### console
 
+This is the simplest notifier. It outputs messages to the console and file log and demonstrates how to implement notification procedures.
+
+##### log_level
+
+type: str, optional, default: DEBUG, values: DEBUG, INFO, WARNING...
+
+Set log level of notification messages.
+
 #### html
 
+Output test results in html format. It can be used for local development or as the body of an email.
+
+##### path
+
+type: str, optional, default: lisa.html
+
+Specify the output file name and path.
+
+##### auto_open
+
+type: bool, optional, default: False
+
+When set to True, the html will be opened in the browser after completion. Useful in local run.
+
+Example of notifier:
+```
+notifier:
+  - type: console
+    log_level: INFO
+  - type: html
+    path: .\lisa.html
+    auto_open: true
+```
 
 ### environment
 
-#### name
+List of environments. For more info, refer [node and environment](https://github.com/microsoft/lisa/blob/main/docs/concepts.md#node-and-environment).
+
+#### environments
+
+Define test run environment.
+
+##### name
+
+type: str, optional, default is empty
 
-#### topology
+Specify the name of the environment.
 
-#### nodes
+##### topology
 
-#### nodes_requirement
+type: str, optional, default is "subnet"
 
-##### type
+Specify the topology of the environment.
+
+##### nodes
+
+List of node, it can be a virtual machine on Azure or Hyper-V, bare metal or others. For more info, refer [node and environment](https://github.com/microsoft/lisa/blob/main/docs/concepts.md#node-and-environment).
+
+##### nodes_requirement
+
+Testing required environment, e.g. node_count (default is 1), core_count (default is 1), memory_mb (default is 512 MB), disk_count (default is 1) etc. The node can be created once the node requirement is met.
+
+###### type
+
+type: str, optional, default value is "requirement", supported values are "requirement", "remote", "local".
```

Comment:
Explain what's difference between them. And how to use for each of them. Especially the remote, it's must have in ready environment.